larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+};
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline size_t DiscreteProbabilityVector::GetSize() const
+{
+    return m_discreteProbabilityData.size();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::GetX(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetX();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::GetProbability(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetDensityDatum()*(m_useWidths ? m_discreteProbabilityData.at(index).GetWidth() : 1.f);
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::GetProbabilityDensity(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetDensityDatum();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::GetCumulativeProbability(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetCumulativeDatum();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline float DiscreteProbabilityVector::GetWidth(const size_t index) const
+{
+    this->VerifyElementRequest(index);
+
+    return m_discreteProbabilityData.at(index).GetWidth();
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline void DiscreteProbabilityVector::GetAllAtIndex(const size_t index, float &x, float &probabilityDensity,
+    float &cumulativeProbability, float &width) const
+{
+    this->VerifyElementRequest(index);
+
+    const DiscreteProbabilityDatum &theDatum(m_discreteProbabilityData.at(index));
+    x = theDatum.GetX();
+    probabilityDensity = theDatum.GetDensityDatum();
+    cumulativeProbability = theDatum.GetCumulativeDatum();
+    width = theDatum.GetWidth();
+
+    return;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+inline DiscreteProbabilityVector::DiscreteProbabilityDatum::DiscreteProbabilityDatum(const float x,
+    const float densityDatum, const float cumulativeDatum, const float width) :
```

Comment:
Extra 4 spaces indent for constructor arguments.

Reply:
More spaces!